### DISCUSSION TITLE
how to binding event to devexpress  treelist control?

### DISCUSSION THREAD

**Discussion by chenhaoxianshen** [Caliburn-Micro/Caliburn.Micro#810] [General]
cal:Message.Attach="[Event FocusedRowChangedEvent]=[Action RowValueChanged]"

why it doesn't work?

**Comment by mvermef** [comment]
Can you show us the relavent xaml... You might have the datacontext incorrect.

**Comment by vb2ae** [comment]
The FocusedRowChangedEvent is obsolete use the CurrentItemChanged event instead.  I included the eventArgs being passed to event handler

Important Xaml

               <dxt:TreeListControl ItemsSource="{Binding employees}" cal:Message.Attach="[Event CurrentItemChanged]=[Action RowValueChanged($eventArgs)]">


Code I used for event handler

        public void RowValueChanged(CurrentItemChangedEventArgs eventArgs )
        {
            var newEmployee = eventArgs.NewItem as Employee;
            Trace.WriteLine(newEmployee.Name);
        }


[https://docs.devexpress.com/WPF/DevExpress.Xpf.Grid.DataViewBase.FocusedRowChanged](https://docs.devexpress.com/WPF/DevExpress.Xpf.Grid.DataViewBase.FocusedRowChanged)
